src/api/middlewares/restrict.middleware.js

```diff
@@ -0,0 +1,15 @@
+import { StatusCodes } from "http-status-codes";
+
+const restrictTo = (...roles) => {
+  return (req, res, next) => {
+    const role = req.user.role;
+    if (!roles.includes(role.toUpperCase())) {
+      return res
+        .status(StatusCodes.UNAUTHORIZED)
+        .json({ message: "You are not authorized to perform this action" });
+    }
+    next();
+  };
+};
```

Comment:
add checkUserLoggedIn and restrictTo into same file `protect.middleware.js`

Reply:
> I can't see checkLoggedIn and restrict middlewares in roles routes

Check in the index, to increase code smell by reducing redundancy those middleware are used on the primary route of roles in index. Check well.